### PR TITLE
cmake, doc: Update `doc/productivity.md`

### PR DESCRIPTION
**Pull request by hebasto**

### CODE REVIEW

**Comment by hebasto** [comment]
cc @maflcko @paplorinc

File: doc/productivity.md [hunk 1]
```diff
@@ -46,38 +46,33 @@ Note: base_dir is required for ccache to share cached compiles of the same file
 
 You _must not_ set base_dir to "/", or anywhere that contains system headers (according to the ccache docs).
 
-### Disable features with `./configure`
+### Disable features when generating the buildsystem
 
-After running `./autogen.sh`, which generates the `./configure` file, use `./configure --help` to identify features that you can disable to save on compilation time. A few common flags:
+During generating the buildsystem, only essential build options are enabled by default to save on compilation time.
```
Comment (paplorinc) [inline doc/productivity.md:51]:
```suggestion
During the generation of the build system only essential build options are enabled by default to save on compilation time.
```

File: doc/productivity.md [hunk 2]
```diff
@@ -46,38 +46,33 @@ Note: base_dir is required for ccache to share cached compiles of the same file
 
 You _must not_ set base_dir to "/", or anywhere that contains system headers (according to the ccache docs).
 
-### Disable features with `./configure`
+### Disable features when generating the buildsystem
```
Comment (paplorinc) [inline doc/productivity.md:49]: build system is usually separated in existing code

File: doc/productivity.md [hunk 3]
```diff
@@ -46,38 +46,33 @@ Note: base_dir is required for ccache to share cached compiles of the same file
...
-### Make use of your threads with `make -j`
+### Make use of your threads with `-j`
 
-If you have multiple threads on your machine, you can tell `make` to utilize all of them with:
+If you have multiple threads on your machine, you can utilize all of them with:
 
 ```sh
-make -j"$(($(nproc)+1))"
+cmake --build build -j "$(($(nproc)+1))"
+ctest --test-dir build -j "$(($(nproc)+1))"
```
Comment (paplorinc) [inline doc/productivity.md:63]: I doubt anyone will use +1 part, I vote for simplifying, like in https://github.com/bitcoin/bitcoin/pull/30619/files#diff-fb3fb3681e7dbfc61c47a7d1f890941b8c886cd46fc06cf7839a03a68dc2aa02L68

File: doc/productivity.md [hunk 4]
```diff
@@ -46,38 +46,33 @@ Note: base_dir is required for ccache to share cached compiles of the same file
...
 
-### Make use of your threads with `make -j`
+### Make use of your threads with `-j`
 
-If you have multiple threads on your machine, you can tell `make` to utilize all of them with:
+If you have multiple threads on your machine, you can utilize all of them with:
 
 ```sh
-make -j"$(($(nproc)+1))"
+cmake --build build -j "$(($(nproc)+1))"
```
Comment (paplorinc) [inline doc/productivity.md:62]: let's use the comment without mentioning nproc

**Review by paplorinc — COMMENTED** [review]
Quick first pass

File: doc/productivity.md (hunk 1 above)
Comment (hebasto) [reply doc/productivity.md]: Thanks! Reworked.

File: doc/productivity.md (hunk 2 above)
Comment (hebasto) [reply doc/productivity.md]: Thanks! Updated.

File: doc/productivity.md (hunk 3 above)
Comment (hebasto) [reply doc/productivity.md:63]: In this PR, I don't want introduce any changes that are unrelated to the migration to CMake.

File: doc/productivity.md (hunk 4 above)
Comment (hebasto) [reply doc/productivity.md:62]: In this PR, I don't want introduce any changes that are unrelated to the migration to CMake.

Comment (paplorinc) [reply doc/productivity.md:62]: ok, but you've added an extra nproc, that's what I was commenting on

File: doc/productivity.md [hunk 5]
```diff
@@ -46,38 +46,33 @@ Note: base_dir is required for ccache to share cached compiles of the same file
...
 
 Obviously, it is important to build and run the tests at appropriate times -- but when you just want a quick compile to check your work, consider picking one or a set of build targets relevant to what you're working on, e.g.:
 
 ```sh
-make src/bitcoind src/bitcoin-cli
-make src/qt/bitcoin-qt
-make -C src bitcoin_bench
+cmake --build build -t bitcoind bitcoin-cli
+cmake --build build -t bitcoin-qt
+cmake --build build -t bench_bitcoin
```
Comment (paplorinc) [inline doc/productivity.md:75]: nit: we seem to be using `--target` mor often in code

**Review by paplorinc — APPROVED** [review]
ACK 320566407b3c36b187df38b0dfbc6e1ba18e2972

File: doc/productivity.md (hunk 5 above)
Comment (hebasto) [reply doc/productivity.md]: Thanks! Reworked.

File: doc/productivity.md (hunk 4 above)
Comment (hebasto) [reply doc/productivity.md:62]: The extra line has been dropped.

**Comment by paplorinc** [comment]
ACK 8601cea1e9b78dc9fa01318eb58b2313aa1165be

**Pull request merged**